games/xworld3d/maps/xworld3d_env.py

```diff
@@ -311,49 +314,125 @@ def _configure(self):
+                                             "id", \
+                                             "loc", \
+                                             "asset_path", \
+                                             "yaw"])
+        pv_dict = default_dict.copy()
+        pv_dict.update(property_value_dict)
+
+        # pre-processing for name and asset_path due to their dependency
+        path_value = property_value_dict.get("asset_path", "empty")
+        name_value = property_value_dict.get("name", "empty")
+
+        # if name is specified by user but asset_path is unspecified
+        if name_value != "empty" and path_value == "empty":
+            property_value_dict["asset_path"] = None
+            property_value_dict["id"] = property_value_dict.get("id")
+
+        # if asset_path is specified
+        if path_value != "empty":
+            # if name is not specified then derive name from asset_path
+            if name_value == "empty" and path_value != None:
+                names = [n for n in self.items[entity.type] if path_value in set(self.items[entity.type][n])]
+                assert len(names) == 1, "there should be only one name corresponding to the asset_path: %s" % (path_value)
+                property_value_dict["name"] = names[0]
+                property_value_dict["id"] = property_value_dict.get("id")
+                pv_dict.update(property_value_dict)
+            elif name_value == "empty" and path_value == None:
+                property_value_dict["name"] = None
+                property_value_dict["id"] = property_value_dict.get("id")
+            else:
+                # if both name and asset_path are specified, check consistency
+                assert path_value in self.items[entity.type][name_value], \
+                    "specified name: %s and asset_path: %s mis-match" % (name_value, path_value)
+
+        for property in pv_dict:
+            assert property in entity.__dict__.keys() and property in default_dict.keys(), \
+                "invalid property name: %s is provided" % property
+            value = pv_dict[property]
+
+            # skip unspecified and non-empty entity properties
+            if property not in property_value_dict.keys() and entity.__dict__[property] is not None:
+                continue
+
+            if property == "loc":
+                if entity.loc is not None:
+                    self.available_grids.append(entity.loc)
+                entity.loc = check_or_get_value(value, self.available_grids)
+                self.available_grids.remove(entity.loc)
+            if property == "name":
+                entity.name = check_or_get_value(value, self.get_all_possible_names(entity.type))
+                # update id once name is changed
+                # entity.id = "%s_%s" % (entity.name, self.entity_nums[entity.type])
+            if property == "id":
+                entity.id = "%s_%s" % (entity.name, check_or_get_value(value, range(0, 10000)))
```

Comment:
range(0, 10000) is dangerous. There might be a collision for two different entities of the same class. You can maintain a global running_id for the class.

Reply:
changed with running_id